src/ir/derive.rs

```diff
@@ -90,16 +90,16 @@ pub trait CanDeriveCopy<'a> {
 /// to be a recursive method that checks whether all the proper members can
 /// derive default or not, because of the limit rust has on 32 items as max in the
 /// array.
-pub trait CanDeriveDefault {
+pub trait CanDeriveDefault<'a> {
     /// Implementations can define this type to get access to any extra
     /// information required to determine whether they can derive `Default`. If
     /// extra information is unneeded, then this should simply be the unit type.
     type Extra;
 
     /// Return `true` if `Default` can be derived for this thing, `false`
     /// otherwise.
-    fn can_derive_default(&self,
-                          ctx: &BindgenContext,
+    fn can_derive_default(&'a self,
+                          ctx: &'a BindgenContext,
```

Comment:
These references shouldn't be `'a`; that should just be for `Self::Extra`'s use. Unless they're required and I'm missing why...